src/components/LoginForm/index.tsx

```diff
@@ -1,101 +1,111 @@
-
-  // API
-  const handleLoginClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
-    e.preventDefault();
-    const data = { loginId, password };
-    API.logIn(data);
-  };
-
-  // Id focus
-  const userIdInput = useCallback((inputElement) => {
-    if (inputElement) {
-      inputElement.focus();
-    }
-  }, []);
+  // Valid
+  const [isValidLoginId, setIsValidLoginId] = useState<boolean>(false);
+  const [isValidPassword, setIsValidPassword] = useState<boolean>(false);
+  // Error Message
+  const [loginIdErrorMessage, setLoginIdErrorMessage] = useState<string>(ID_REQUIRE_CHECK); // 추후 중복확인 체크 작업 진행
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState<string>(PW_REQUIRE_CHECK);
 
-  // ID input Change Event
-  const onChangeIdInput = (e: SyntheticEvent<HTMLInputElement>) => {
+  // LoginId Change Event
+  const onChangeLoginId = (e: SyntheticEvent<HTMLInputElement>) => {
     setLoginId(e.currentTarget.value);
-    setIsValidId(false);
+    setIsValidLoginId(false);
   };
 
-  // PW input Change Event
-  const onChangePwInput = (e: SyntheticEvent<HTMLInputElement>) => {
+  // Password Change Event
+  const onChangePassword = (e: SyntheticEvent<HTMLInputElement>) => {
     setPassword(e.currentTarget.value);
-    setIsValidPw(false);
+    setIsValidPassword(false);
   };
 
-  // userID focus out
-  const onBlurIdInput = (e: SyntheticEvent<HTMLInputElement>) => {
-    if (!e.currentTarget.value) {
-      setIsValidId(true);
+  // LoginId onBlur Event
+  const onBlurLoginId = () => {
+    if (!loginId) {
+      setIsValidLoginId(true);
     }
   };
 
-  // userID focus out
-  const onBlurPwInput = (e: SyntheticEvent<HTMLInputElement>) => {
-    if (!e.currentTarget.value) {
-      // Pw empty
-      setPwdMsg(PW_REQUIRE_CHECK);
-      setIsValidPw(true);
+  // Password onBlure Event
+  const onBlurPassword = () => {
+    if (!password) {
+      setIsValidPassword(true);
+      setPasswordErrorMessage(PW_REQUIRE_CHECK);
       return;
     }
 
-    if (password.length <= 8) {
-      setPwdMsg(PW_VALID_CHECK);
-      setIsValidPw(true);
+    if (password.length < 8) {
+      setIsValidPassword(true);
+      setPasswordErrorMessage(PW_VALID_CHECK);
+    }
+  };
+
+  // API
+  const handleLoginClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
+    e.preventDefault();
+    if (isValidLoginId === false && !loginId) {
+      setIsValidLoginId(true);
+    } else if (isValidPassword === false && !password) {
+      setIsValidPassword(true);
+    } else {
+      const data = { loginId, password };
+      const response = API.logIn(data);
+      response
+        .then((res) => {
+          if (res.status === 200) {
+            alert(`${res.data.nickname}님 환영합니다.`);
+          }
+        })
+        .catch((error) => {
+          if (error.code === 'ERR_BAD_REQUEST') {
+            error.message = '아이디/패스워드를 확인해주세요';
+            alert(error.message);
+          }
+        });
     }
```

Comment:
```suggestion
  const siginIn = () => {
   const data = { loginId, password };
      const response = API.logIn(data);
      response
        .then((res) => {
          if (res.status === 200) {
            alert(`${res.data.nickname}님 환영합니다.`);
          }
        })
        .catch((error) => {
          if (error.code === 'ERR_BAD_REQUEST') {
            error.message = '아이디/패스워드를 확인해주세요';
            alert(error.message);
          }
        });
        }
  const handleLoginClick = (e: React.MouseEvent<HTMLButtonElement>): void => {
    e.preventDefault();
    if (isValidLoginId === false && !loginId) {
      setIsValidLoginId(true);
    } else if (isValidPassword === false && !password) {
      setIsValidPassword(true);
    } else {
     SiginIn()
    }
```
api호출하는 함수를 따로 빼는건 어떨까요? 1함수 1기능 원칙
현재 handleLoginClick함수가 하는 일
1. 클릭할 때 일어나는 일
2. 유효성 검사 << 의존성이 있어서 리팩토링 필요
3. api호출 << 함수로 분리 가능

Reply:
네, 추가적으로 쿠키 작업할때 수정 진행하겠습니다.